src/EventDispatcher/EventDispatcherInterface.php

```diff
@@ -23,31 +19,18 @@ public function hasListeners(string $eventName, string $class, string $format):
      * The listeners/subscribers are called in the same order in which they
      * were added to the dispatcher.
      *
-     * @param string $eventName
-     * @param string $class
-     * @param string $format
-     * @param Event $event
-     * @return void
      */
-    public function dispatch($eventName, string $class, string $format, Event $event): void;
+    public function dispatch(string $eventName, string $class, string $format, Event $event): void;
```

Comment:
good catch, this is a bug! :)

Reply:
There is more of missing types, some break more things so I would probably not touch types in first iteration and do them separately, then add proper docblocks for properties etc. (this is especially in extremely bad shape in Serializer).
Then we can also consider PHPStan.